barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_trace.cpp

```diff
@@ -126,15 +170,14 @@ void AvmTraceBuilder::op_add(
 {
     auto clk = static_cast<uint32_t>(main_trace.size()) + 1;
 
-    auto const res = resolve_ind_three(call_ptr, clk, indirect, a_offset, b_offset, dst_offset);
-    bool tag_match = res.tag_match;
+    // Resolve any potential indirects in the order they are encoded in the indirect byte.
+    auto [resolved_a, resolved_b, resolved_c] = resolve_indirects<3>(indirect, { a_offset, b_offset, dst_offset });
```

Comment:
Nice. About naming: resolved_a_addr? a_addr? Or maybe if you call a_offset differently you can call this a_addr? (something similar to what we do in TS)

Reply:
i might save this change for later